### DISCUSSION TITLE
Question about using this theme

### DISCUSSION THREAD

**Discussion by Andre601** [mmistakes/jekyll-theme-basically-basic#112] [Q&A] [answered]
Sorry to ask this here and like this, but can you just fork this repo, rename it to `<username>.github.io` to then host it as a GitHub page?
The explanation of using gem (Whatever that is) seemd a bit weird and I would appreciate to avoid using a separate programm just to host a github page.

**Comment by mmistakes** [comment, marked as answer]
You can, but I wouldn't advise that. Anytime the theme is updated you won't be able to easily merge those changes back in.

If you're hosting on GitHub, follow those instructions in the README. You basically create a repo with a `_config.yml` file and your posts/pages and that's it. The layouts, includes, stylesheets, etc are pulled automatically by GitHub every time the site is built.

Best part you automatically get any theme updates this way too.

If you're looking for a starter, take a look at the [`/docs` folder](https://github.com/mmistakes/jekyll-theme-basically-basic/tree/master/docs). That's the source for the demo site and is built on GitHub Pages using the method I mentioned above. You could copy that folder, update `_config.yml` to your liking, replace `_posts` and `_pages` with your own content, and call it a day.

**Comment by Andre601** [comment]
I currently run a version here: https://github.com/Andre601/Andre601.github.io

I currently face a minor issue where GitHub reports the following message each time I build the page:
```
The page build completed successfully, but returned the following warning for the `master` branch:

You are attempting to use a Jekyll theme, "jekyll-theme-basically-basic", which is not supported by GitHub Pages. Please visit https://pages.github.com/themes/ for a list of supported themes. If you are using the "theme" configuration variable for something other than a Jekyll theme, we recommend you rename this variable throughout your site. For more information, see https://help.github.com/en/github/working-with-github-pages/adding-a-theme-to-your-github-pages-site-using-jekyll.

For information on troubleshooting Jekyll see:

  https://help.github.com/articles/troubleshooting-jekyll-builds

If you have any questions you can contact us by replying to this email.
```

I don't really have that big of interest to keep this updated as I just want to host a site with static and blog pages.

**Comment by mmistakes** [comment]
You need to install the theme following the [GitHub compatible method](https://github.com/mmistakes/jekyll-theme-basically-basic#github-pages-method). GitHub Pages only supports a few 1st party themes using the `theme` property in your _config.yml

If you want to use this theme you need to set `remote_theme: "mmistakes/jekyll-theme-basically-basic"` instead.

**Comment by mmistakes** [comment]
Also if you want to go with a fork you'll need to remove all the junk.
https://github.com/mmistakes/jekyll-theme-basically-basic#remove-the-unnecessary

**Comment by Andre601** [comment]
Is it important where I post/upload files to in the repository?

**Comment by mmistakes** [comment]
Up to you how you manage that. As long as you use a valid path in your content shouldn't matter where you put them.

I tend to put everything in `/assets` or subfolders like `/assets/images/`. Then you can link to them with Markdown or HTML like so:

```markdown
[link to a file](/assets/your-file.pdf)
```

```html
<a href="/assets/your-file.pdf">link to a file</a>
```

**Comment by Andre601** [comment]
I talk about the pages themself.
When looking at the example file does it look like it has to be in the same folder where the _config.yml is right?

**Comment by mmistakes** [comment]
I suggest you start with [Jekyll's documentation](https://jekyllrb.com/) to get a feel for how to use it  and follow how to create posts and files.

This is a reason why I don't advise forking the repo. It's not setup in a way that makes it a good starter. It has extra folders and the demo/example sites aren't setup in the same way you'd do it in your own repo.

**Comment by Andre601** [comment]
Thanks for the information.
